cmd/migrate/main.go

```diff
@@ -0,0 +1,161 @@
+	// failes for whatever reason
+	for _, downstreamMachine := range downstreamMachines.Items {
+		convertedClusterv1alpha1Machine, err := conversions.ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(downstreamMachine)
+		if err != nil {
+			return fmt.Errorf("failed to convert machine %s: %v", downstreamMachine.Name, err)
+		}
+
+		createdClusterV1alpha1Machine, err := clusterv1alpha1Client.ClusterV1alpha1().Machines(convertedClusterv1alpha1Machine.Namespace).Create(convertedClusterv1alpha1Machine)
+		if err != nil {
+			return fmt.Errorf("failed to create clusterv1alpha1.machine %s: %v", convertedClusterv1alpha1Machine.Name, err)
+		}
+
+		node, err := kubeClient.CoreV1().Nodes().Get(convertedClusterv1alpha1Machine.Spec.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Failed to get node %s for machine %s: %v", convertedClusterv1alpha1Machine.Spec.Name, convertedClusterv1alpha1Machine.Name, err)
+		}
+		for idx, ownerRef := range node.OwnerReferences {
+			if ownerRef.UID == downstreamMachine.UID {
+				node.OwnerReferences = append(node.OwnerReferences[:idx], node.OwnerReferences[idx+1:]...)
+				break
+			}
+		}
+		newOwnerRef := node.OwnerReferences
+		gv := clusterv1alpha1.SchemeGroupVersion
+		newOwnerRef = append(newOwnerRef, *metav1.NewControllerRef(createdClusterV1alpha1Machine, gv.WithKind("Machine")))
+
+		// We retry this because nodes get frequently updated so there is a reasonable chance this may fail
+		if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			node, err := kubeClient.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			node.OwnerReferences = newOwnerRef
+			if _, err := kubeClient.CoreV1().Nodes().Update(node); err != nil {
+				return err
+			}
+			return nil
+		}); err != nil {
+			return fmt.Errorf("failed to update OwnerRef on node %s: %v", node.Name, err)
+		}
+
+		finalizers := sets.NewString(downstreamMachine.Finalizers...)
+		finalizers.Delete(controller.FinalizerDeleteInstance)
+		downstreamMachine.Finalizers = finalizers.List()
+		if _, err := downstreamClient.MachineV1alpha1().Machines().Update(&downstreamMachine); err != nil {
+			return fmt.Errorf("failed to update downstream machine %s after removing finalizer: %v", convertedClusterv1alpha1Machine.Name, err)
+		}
+		if err := downstreamClient.MachineV1alpha1().Machines().Delete(convertedClusterv1alpha1Machine.Name, nil); err != nil {
+			return fmt.Errorf("failed to delete machine %s: %v", convertedClusterv1alpha1Machine.Name, err)
+		}
+
+		if err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+			return isDownstreamMachineDeleted(convertedClusterv1alpha1Machine.Name, downstreamClient)
+		}); err != nil {
+			return fmt.Errorf("failed to wait for machine %s to be deleted: %v", convertedClusterv1alpha1Machine.Name, err)
+		}
+	}
+	return nil
```

Comment:
would be nice to inform the operator that migration of the machine has been completed successfully.